packages/kbn-es/src/utils/keystore.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const execa = require('execa');
+const { format } = require('util');
+
+const { ES_KEYSTORE_BIN } = require('../paths');
+
+exports.createKeystore = function(installPath) {
+  const proc = execa(ES_KEYSTORE_BIN, ['create'], {
+    cwd: installPath,
+    input: 'n',
+  });
+
+  return new Promise((resolve, reject) => {
+    proc.stdout.on('data', data => {
+      data.toString().match(/already exists/)
+        ? reject('an Elasticsearch Keystore already exists')
```

Comment:
Sorry, but I'm a stickler about not throwing/rejecting with non-Error values. Maybe we can just resolve with an object that says `{ created: false, reason: 'an Elasticsearch Keystore already exitsts' }`?

Reply:
Also, this won't ever resolve is the proc errors, can't be found, or if the first chunk emitted by stdout doesn't have the but about `already exists`. I suggest we change this to something like

```js
exports.createKeystore = async function(installPath) {
  try {
    await execa(ES_KEYSTORE_BIN, ['create'], {
      cwd: installPath,
      input: 'n'
    });
    return { created: true };
  } catch (error) {
    if (error.stdout && error.stdout.match(/already exists/)) {
      return { created: false, reason: 'an Elasticsearch Keystore already exists' }
    }

    throw error
  }
};
```